### ISSUE TITLE
None stop "Checking for messages..." notification 

### ISSUE THREAD

**Issue by h-n-a** [signalapp/Signal-Android#12904]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
This issue has been reported 2 years ago and it started happening again 2 releases back:
https://github.com/signalapp/Signal-Android/issues/11071

I get bombarded none stop while listening to music, browsing the internet, typing a note. basically, anything.

### Steps to reproduce
The issue happens after upgrading to the latest signal app (since the last 2-3 releases)
No specific steps to reproduce it.

### Screenshots

![signalapp](https://user-images.githubusercontent.com/126036770/232262685-d0f5075c-2e1b-4121-8c3c-6a3872364dfa.PNG)

### Device info

**Device:** Oneplus Nord N10 5G (BE2029)
**Android version:** 11.0.12.BE86AA
**Signal version:** 6.17.3

### Link to debug log

[signal-log-1681608303986.zip](https://github.com/signalapp/Signal-Android/files/11241076/signal-log-1681608303986.zip)

**Comment by h-n-a** [comment]
bueller?

**Comment by h-n-a** [comment]
@cody-signal only that message. sometimes it's about an actual notification. like someone adding emoji to a comment or just "checking for messages..." without an actual new message or interaction.

it's very inconsistent and becoming a nuisance.

**Comment by cody-signal** [comment]
> it's very inconsistent and becoming a nuisance.

I feel ya, unfortunately this is currently by design to ensure proper message delivery. Android requires us to post a notification in order to get foreground execution time when we receive a push notification.

However, you should almost always see a message notification afterwards unless most of the messages are in muted/blocked chats.

**Comment by h-n-a** [comment]
lol, what? why did this only happen recently?
If you don't want to address the issue. that's fine. but this is something recent, not "by design". 
Your team forces users to update the software from working versions to untested code just to bloat the app with unnecessary features. trying to make it like whatsapp.

**Comment by neirbowj** [comment]
@cody-signal said,
> Android requires us to post a notification in order to get foreground execution time when we receive a push notification.

Ok, so why does the notification persist for minutes at a time? ... while another person on an iPhone has already received the common message? What are you doing during that "foreground execution time"? Mining cryptocurrency!?

This has been going on across multiple app updates. These messages are definitely not muted both blocked. I *want* these messages promptly at almost any hour of the day or night. The app release notes often talk about new emoji and make claims such as "• Go fast – Messages are delivered quickly and reliably, even on slow networks. Signal is optimized to operate in the most constrained environment possible." I have a greater-than 100Mbps fiber ISP, and messages are often delayed. I associate the "Checking for messages..." notification with new Signal app versions on Android. Should I get off Android? Or get off Signal?

**Comment by greyson-signal** [comment]
@h-n-a 
> but this is something recent, not "by design".


It's both recent and by design. A lot of people have issues receiving Signal notifications because of various OEM battery optimization policies, and one of the things we're trying is starting a foreground service when we receive the push to give us a more reliable execution window to fetch messages. 

@neirbowj 
> Ok, so why does the notification persist for minutes at a time?

Not sure, but guessing that your device is restricting our network access, even when we receive a high-priority push and have a foreground service active. If you share a link to a [debuglog](https://support.signal.org/hc/en-us/articles/360007318591-Debug-Logs-and-Crash-Reports#android_debug), I can look and verify that theory. 

> Should I get off Android? Or get off Signal?

I understand the frustration. We get frustrated too, because getting notifications to work reliably across all OEMs is a real struggle (can read more [here](https://dontkillmyapp.com/)). Many popular messaging applications are "whitelisted" on some OEMs so that they'll always have their notifications delivered, while other apps get heavily throttled in the name of battery optimization. Since we're not on those lists, we have to try a lot of different approaches, and we're continually working on more. If you could share a debuglog that would really help us so we can look into the specific issue you're having.

**Comment by neirbowj** [comment]
@greyson-signal I captured this [debug log](https://debuglogs.org/android/6.27.10/011d15eef2b10b916dc315fac8ac57b0f97f3feb39abbc270fb3621bde08af77) while the "Checking for messages..." notification was active.

**Comment by h-n-a** [comment]
I noticed that the trigger happens when i try to check the desktop client on my muted group chats, it would trigger that immediately. 
Something has been changed 100% and it wasn't by design. 
And it also gets triggered while reactions/new comments are being made on my muted rooms.

**Comment by skicek** [comment]
I start noticing this issue also android 14 clean one, and it is spinning for a long time without any new message after. This happens after I disable battery optimalization for Singal. With enabled it spin for shorter time

> It's both recent and by design. A lot of people have issues receiving Signal notifications because of various OEM battery optimization policies, and one of the things we're trying is starting a foreground service when we receive the push to give us a more reliable execution window to fetch messages.

I guess that would explain battery drain that I notice recently more details here https://github.com/signalapp/Signal-Android/issues/12341#issuecomment-2017856655

**Comment by zilexa** [comment]
Same here! Super annoying. Even with Signal app open, I'm looking at my messages. Then that notification comes and it takes 30 seconds or longer before I actually see the new message. It makes no sense and if I need an urgent reply I nowadays switch more often to WhatsApp because of this.

Android 14 on Asus ZenFone 9.
Whether I have battery usage set to "optimized" (default) and background app management enabled (default), or set the former to "unrestricted" and the latter to disabled doesn't seem to matter.
For WhatsApp I never changed these values.

**Comment by regularroutine** [comment]
Android 14, Asus ZenFone 10, just switched to this phone and I am having exact same issue as zilexa. Didnt happen on my old Pixel phone.

**Comment by greyson-signal** [comment]
Hey folks, for people who are newly experiencing this, can you please post a [debuglog](https://support.signal.org/hc/en-us/articles/360007318591-Debug-Logs-and-Crash-Reports#android_debug)? Otherwise we can't do much. Thank you!

**Comment by regularroutine** [comment]
https://debuglogs.org/android/7.2.4/f3dd16127917c659247d8a6520c1a08d87b31652fa7983a1a47eae710e9ca2c8

**Comment by jcrben** [comment]
https://debuglogs.org/android/7.8.1/3b454d7950f24a233ed758a0f59feef1e555c653ddd6e0edfdb8fb4811b668cb

**Comment by jugarrit** [comment]
https://debuglogs.org/android/7.10.3/d1e48eccb288bf129a6ffcd24ca0b5b442f20cce4b5cbed8ed71c7e50186a5e6

**Comment by hegerj** [comment]
https://debuglogs.org/android/7.11.3/83622cbe91760bd9cd8c7c6782ac5d13bba9b23419e28468957acf3f463c6bac

I don't notice them on my phone (they disappear almost immediately under normal circumstances), however my smartwatch catches them and alerts me all the time.

Also I noticed that the notification appears far more often when I actually Mute a conversation. In that case I get no notification about new messages but loads of "Checking for messages..." on changes (not only on new messages but also on emoji reactions).

**Comment by zilexa** [comment]
Everyone I know in Netherlands already stopped using Signal because of this annoyance. I made such a huge effort for years to get family and friends to move.
I give up. Millions of dollars of donations and not a single native Android app developer that can solve this year's old issue or even take a moment to report back here. 

No more Signal for me.

**Comment by vandervecken** [comment]
I have been seeing this also. It is coinciding with slow message delivery to my Android (Pixel 6a) but not to my desktop app.
The notification lasts for a minute or so, and the message arrives at the *end*, not the beginning.
Debuglog:
https://debuglogs.org/android/7.12.3/ec062f0c233a6fe5e746c33bebe8cf4e9257d9fabcb3bd380d3ad25ec88cf359

**Comment by nauvillain** [comment]
This happens on my Android 14 (Oneplus 12) regularly, too. Mostly when I use the desktop version. I think it happens when I am idle on the desktop version - I get a "Checking for messages" from Signal. I wish I could disable that, as I check my watch every time, only to see that there is no new message.

**Comment by omartijn** [comment]
Not sure why, but in the latest version this has gotten _way more annoying_. I now get a notification _every time I minimize signal_.

**Comment by omartijn** [comment]
I've upgraded to 7.14. So far haven't seen any bogus notifications.

**Comment by hegerj** [comment]
Unfortunately I am still encountering the issue.

https://debuglogs.org/android/7.15.4/112a102eb809ace4984914e26d19dfb2627f7f12572ea82d1e6c25a46e5b762f

**Comment by cody-signal** [comment]
@hegerj and this happens even after opening Signal? It looks like you have Signal setup to get demoted by the OS for power reasons which could be why you're getting the message in the first place. But if it doesn't go away after opening the app that's a whole other issue.

**Comment by hegerj** [comment]
I turned off any power optimization I could find, no change in behavior - more to the point I rarely *not* get notification on time, so the app seems to be running in background fine.

**Comment by twothreenine** [comment]
Would it be possible to change the symbol for "checking for messages" so it is distinguishable from significant ones like "new message" etc.? The backup notification has its own too.

This would help massively when the icon appears on the lock screen (in case only icons are shown) or on the top row.

(Android.)

**Comment by GertBurger** [comment]
Can one change the settings of the notification channel? e.g. set it to silent without showing on lock screen..
I can't seem to find a notification channel that affects this notification

**Comment by emdej** [comment]
Could we just have an option in settings to disable it at all? v7.38.7
Frequently getting notifications on smartwatch. Not visible on smartphone, but smartwatch vibrates and shows notifications...

![Image](https://github.com/user-attachments/assets/5a58e215-7c69-4ebd-886d-e9f6155ac075)

**Comment by emdej** [comment]
I've disabled notifications from category "Other". I'll let you know if it doesn't help.

**Comment by kees** [comment]
I think this happens for messages going to Muted chats. My watch buzzes for every single one of those messages (with the "Checking for messages..." text), but my phone does _NOT_ buzz, because it's a message for a Muted chat.

And thank you @emdej turning off "Other" totally silenced it for me. *WHEW*

**Comment by itimgit** [comment]
It seems to happen to archived and muted chats. Also my wearable keeps getting confused by this. Disable all signal notifications, still no luck. Fortunately wearable allowed per-app setting, now at least my wrist isn't constantly buzzing for nothing anymore.

**Comment by GertBurger** [comment]
On android you can also disallow access to silenced notifications on a per app basis, I use this with my Garmin so that my watch doesn't get silenced notifications

**Comment by roelvanklink** [comment]
I fixed this by clearing signal's cache in under Apps and restarting my phone

**Comment by greyson-signal** [comment]
Yeah, this is in the "Other" notification channel. We disable vibrate, but some phones don't like having channels with vibration enabled, so it gets a little weird. The 7.46 release will contain a change to ensure the icon for these messages is distinct. Thank you!

**Comment by any1** [comment]
I've had this problem with my phone since this morning when I set the "chat length limit to 1000" because signal was filling up my storage.

**Comment by keysmashes** [comment]
It would be *really* handy if these could have their own notification channel. "Other" sounds like it could plausibly be used for important/interesting things (if not now, then maybe in the future) which I potentially would prefer not to disable. Something like "Background activities" or even just "Checking for messages" would make it really obvious that nothing would be missed by disabling these notifications.

**Comment by yardenac** [comment]
Yes please! I came here just to advocate for giving these their own notification channel that can be turned off.
